stdlib/public/core/UnsafeBufferPointer.swift.gyb

```diff
@@ -302,26 +354,109 @@ extension Unsafe${Mutable}BufferPointer: ${Mutable}Collection, RandomAccessColle
     return result.partialValue
   }
 
-  @inlinable // unsafe-performance
-  public func _failEarlyRangeCheck(_ index: Int, bounds: Range<Int>) {
-    // NOTE: In release mode, this method is a no-op for performance reasons.
-    _debugPrecondition(index >= bounds.lowerBound)
-    _debugPrecondition(index < bounds.upperBound)
+  /// Accesses the element at the specified position.
+  ///
+%if Mutable:
+  /// The following example uses the buffer pointer's subscript to access and
+  /// modify the elements of a mutable buffer pointing to the contiguous
+  /// contents of an array:
+  ///
+  ///     var numbers = [1, 2, 3, 4, 5]
+  ///     numbers.withUnsafeMutableBufferPointer { buffer in
+  ///         for i in stride(from: buffer.startIndex, to: buffer.endIndex - 1, by: 2) {
+  ///             let x = buffer[i]
+  ///             buffer[i + 1] = buffer[i]
+  ///             buffer[i] = x
+  ///         }
+  ///     }
+  ///     print(numbers)
+  ///     // Prints "[2, 1, 4, 3, 5]"
+  ///
+  ///     Uninitialized memory cannot be initialized to a nontrivial type
+  ///     using this subscript. Instead, use an initializing method, such as
+  ///     `initializeElement(at:to:)`
+%else:
+  /// The following example uses the buffer pointer's subscript to access every
+  /// other element of the buffer:
+  ///
+  ///     let numbers = [1, 2, 3, 4, 5]
+  ///     let sum = numbers.withUnsafeBufferPointer { buffer -> Int in
+  ///         var result = 0
+  ///         for i in stride(from: buffer.startIndex, to: buffer.endIndex, by: 2) {
+  ///             result += buffer[i]
+  ///         }
+  ///         return result
+  ///     }
+  ///     // 'sum' == 9
+%end
+  ///
+  /// - Note: Bounds checks for `i` are performed only in debug mode.
+  ///
+  /// - Parameter i: The position of the element to access. `i` must be in the
+  ///   range `0..<count`.
+  @_alwaysEmitIntoClient
+  public subscript(i: Int) -> Element {
+    @_transparent
```

Comment:
I'm pretty sure we've changed it to allow attributes on the var decl to be applied to the accessors as well, so we can mark this `@_transparent` once. Testing this would be useful.

Reply:
No time! 😛 We can do that work later, or just leave it be.